File: backend/laravel/app/Models/Appointment.php
<?php

namespace App\Models;
use Illuminate\Database\Eloquent\Model;

class Appointment extends Model
{
    protected $fillable = [
        'id_schedule',
        'id_patient',
        'name_patient',
        'date',
        'status',
        'start_time',
        'created_by',
        'updated_by',
    ];

    // Definición de las relaciones
    public function schedule()
    {
        return $this->belongsTo(Schedule::class, 'id_schedule');
    }

    public function patient()
    {
        return $this->belongsTo(User::class, 'id_patient');
    }
}

File: frontend/lib/features/calendar/domain/entities/appointment.dart
class Appointment {
  final int id;
  final int idSchedule;
  final DateTime date;
  final String startTime;
  final String status;
  final String patientName;
  final String clinicName;
  final int? doctorId;
  final String? doctorName;

  Appointment({
    required this.id,
    required this.idSchedule,
    required this.date,
    required this.startTime,
    required this.status,
    required this.patientName,
    required this.clinicName,
    this.doctorId,
    this.doctorName,
  });
}
